Error when compiling for NW.js 0.38.x (on OSX)

Hello,

compiling on NW.js 0.38.x it fails the error shows maybe a change in v8 headers?
i'm on OSX i didn't tested on other operating systems
compiling for 0.37.x was working

`../src/api/greenworks_api_utils.cc:69:38: error: too few arguments to function call, single argument 'context' was not specified
  constructor.Reset(tpl->GetFunction());
                    ~~~~~~~~~~~~~~~~ ^
/Users/marco/.nw-gyp/0.38.0/deps/v8/include/v8.h:5993:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/marco/.nw-gyp/0.38.0/deps/v8/include/v8config.h:347:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/api/greenworks_api_utils.cc:70:74: error: too few arguments to function call, single argument 'context' was not specified
  Nan::Set(exports, Nan::New("Utils").ToLocalChecked(), tpl->GetFunction());
                                                        ~~~~~~~~~~~~~~~~ ^
/Users/marco/.nw-gyp/0.38.0/deps/v8/include/v8.h:5993:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
  ^
/Users/marco/.nw-gyp/0.38.0/deps/v8/include/v8config.h:347:31: note: expanded from macro 'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
2 errors generated.
`

I

--- a/package.json
+++ b/package.json
@@ -13,7 +13,7 @@
     "compdb": "node-gyp configure -- -f ninja && ninja -C build/Release -t compdb cxx > compile_commands.json"
   },
   "dependencies": {
-    "nan": "^2.8.0"
+    "nan": "^2.14.0"
   },
   "devDependencies": {
     "mocha": "1.21.4"

--- a/src/api/greenworks_api_utils.cc
+++ b/src/api/greenworks_api_utils.cc
@@ -66,8 +66,9 @@ void RegisterAPIs(v8::Local<v8::Object> exports) {
   Nan::SetMethod(tpl, "createArchive", CreateArchive);
   Nan::SetMethod(tpl, "extractArchive", ExtractArchive);
   Nan::Persistent<v8::Function> constructor;
-  constructor.Reset(tpl->GetFunction());
-  Nan::Set(exports, Nan::New("Utils").ToLocalChecked(), tpl->GetFunction());
+  constructor.Reset(Nan::GetFunction(tpl).ToLocalChecked());
+  Nan::Set(exports, Nan::New("Utils").ToLocalChecked(),
+           Nan::GetFunction(tpl).ToLocalChecked());
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_achievement.cc
+++ b/src/api/steam_api_achievement.cc
@@ -86,22 +86,11 @@ NAN_METHOD(GetNumberOfAchievements) {
 }
 
 void RegisterAPIs(v8::Local<v8::Object> target) {
-  Nan::Set(target,
-           Nan::New("activateAchievement").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(ActivateAchievement)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getAchievement").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetAchievement)->GetFunction());
-  Nan::Set(target,
-           Nan::New("clearAchievement").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(ClearAchievement)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getAchievementNames").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetAchievementNames)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getNumberOfAchievements").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               GetNumberOfAchievements)->GetFunction());
+  SET_FUNCTION("activateAchievement", ActivateAchievement);
+  SET_FUNCTION("getAchievement", GetAchievement);
+  SET_FUNCTION("clearAchievement", ClearAchievement);
+  SET_FUNCTION("getAchievementNames", GetAchievementNames);
+  SET_FUNCTION("getNumberOfAchievements", GetNumberOfAchievements);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_auth.cc
+++ b/src/api/steam_api_auth.cc
@@ -152,36 +152,14 @@ void RegisterAPIs(v8::Local<v8::Object> target) {
   Nan::Set(target,
            Nan::New("EncryptedAppTicketSymmetricKeyLength").ToLocalChecked(),
            Nan::New(k_nSteamEncryptedAppTicketSymmetricKeyLen));
-  Nan::Set(target,
-           Nan::New("getAuthSessionTicket").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetAuthSessionTicket)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getEncryptedAppTicket").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               GetEncryptedAppTicket)->GetFunction());
-  Nan::Set(target,
-           Nan::New("decryptAppTicket").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               DecryptAppTicket)->GetFunction());
-  Nan::Set(target,
-           Nan::New("isTicketForApp").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               IsTicketForApp)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getTicketIssueTime").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               getTicketIssueTime)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getTicketSteamId").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               getTicketSteamId)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getTicketAppId").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               getTicketAppId)->GetFunction());
-  Nan::Set(target,
-           Nan::New("cancelAuthTicket").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(CancelAuthTicket)->GetFunction());
+  SET_FUNCTION("getAuthSessionTicket", GetAuthSessionTicket);
+  SET_FUNCTION("getEncryptedAppTicket", GetEncryptedAppTicket);
+  SET_FUNCTION("decryptAppTicket", DecryptAppTicket);
+  SET_FUNCTION("isTicketForApp", IsTicketForApp);
+  SET_FUNCTION("getTicketIssueTime", getTicketIssueTime);
+  SET_FUNCTION("getTicketSteamId", getTicketSteamId);
+  SET_FUNCTION("getTicketAppId", getTicketAppId);
+  SET_FUNCTION("cancelAuthTicket", CancelAuthTicket);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_cloud.cc
+++ b/src/api/steam_api_cloud.cc
@@ -175,37 +175,16 @@ NAN_METHOD(GetFileNameAndSize) {
 }
 
 void RegisterAPIs(v8::Local<v8::Object> target) {
-  Nan::Set(target,
-           Nan::New("saveTextToFile").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(SaveTextToFile)->GetFunction());
-  Nan::Set(target,
-           Nan::New("deleteFile").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(DeleteFile)->GetFunction());
-  Nan::Set(target,
-           Nan::New("readTextFromFile").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(ReadTextFromFile)->GetFunction());
-  Nan::Set(target,
-           Nan::New("saveFilesToCloud").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(SaveFilesToCloud)->GetFunction());
-  Nan::Set(target,
-           Nan::New("isCloudEnabled").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(IsCloudEnabled)->GetFunction());
-  Nan::Set(target,
-           Nan::New("isCloudEnabledForUser").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               IsCloudEnabledForUser)->GetFunction());
-  Nan::Set(target,
-           Nan::New("enableCloud").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(EnableCloud)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getCloudQuota").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetCloudQuota)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getFileCount").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetFileCount)->GetFunction());
-  Nan::Set(target,
-           Nan::New("getFileNameAndSize").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetFileNameAndSize)->GetFunction());
+  SET_FUNCTION("saveTextToFile", SaveTextToFile);
+  SET_FUNCTION("deleteFile", DeleteFile);
+  SET_FUNCTION("readTextFromFile", ReadTextFromFile);
+  SET_FUNCTION("saveFilesToCloud", SaveFilesToCloud);
+  SET_FUNCTION("isCloudEnabled", IsCloudEnabled);
+  SET_FUNCTION("isCloudEnabledForUser", IsCloudEnabledForUser);
+  SET_FUNCTION("enableCloud", EnableCloud);
+  SET_FUNCTION("getCloudQuota", GetCloudQuota);
+  SET_FUNCTION("getFileCount", GetFileCount);
+  SET_FUNCTION("getFileNameAndSize", GetFileNameAndSize);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_dlc.cc
+++ b/src/api/steam_api_dlc.cc
@@ -27,7 +27,7 @@ NAN_METHOD(IsDLCInstalled) {
   info.GetReturnValue().Set(SteamApps()->BIsDlcInstalled(dlc_app_id));
 }
 
-NAN_METHOD(installDLC) {
+NAN_METHOD(InstallDLC) {
   Nan::HandleScope scope;
   if (info.Length() < 1 || !info[0]->IsUint32()) {
     THROW_BAD_ARGS("Bad arguments");
@@ -37,7 +37,7 @@ NAN_METHOD(installDLC) {
   info.GetReturnValue().Set(Nan::Undefined());
 }
 
-NAN_METHOD(uninstallDLC) {
+NAN_METHOD(UninstallDLC) {
   Nan::HandleScope scope;
   if (info.Length() < 1 || !info[0]->IsUint32()) {
     THROW_BAD_ARGS("Bad arguments");
@@ -48,14 +48,10 @@ NAN_METHOD(uninstallDLC) {
 }
 
 void RegisterAPIs(v8::Local<v8::Object> target) {
-  Nan::Set(target, Nan::New("getDLCCount").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetDLCCount)->GetFunction());
-  Nan::Set(target, Nan::New("isDLCInstalled").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(IsDLCInstalled)->GetFunction());
-  Nan::Set(target, Nan::New("installDLC").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(installDLC)->GetFunction());
-  Nan::Set(target, Nan::New("uninstallDLC").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(uninstallDLC)->GetFunction());
+  SET_FUNCTION("getDLCCount", GetDLCCount);
+  SET_FUNCTION("isDLCInstalled", IsDLCInstalled);
+  SET_FUNCTION("installDLC", InstallDLC);
+  SET_FUNCTION("uninstallDLC", UninstallDLC);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_friends.cc
+++ b/src/api/steam_api_friends.cc
@@ -266,39 +266,22 @@ NAN_METHOD(GetFriendMessage) {
   info.GetReturnValue().Set(result);
 }
 
-void RegisterAPIs(v8::Local<v8::Object> exports) {
-  InitFriendFlags(exports);
-  InitFriendRelationship(exports);
-  InitFriendPersonaChange(exports);
-  InitAccountType(exports);
-  InitChatEntryType(exports);
+void RegisterAPIs(v8::Local<v8::Object> target) {
+  InitFriendFlags(target);
+  InitFriendRelationship(target);
+  InitFriendPersonaChange(target);
+  InitAccountType(target);
+  InitChatEntryType(target);
 
-  Nan::Set(exports,
-           Nan::New("getFriendCount").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetFriendCount)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getFriends").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetFriends)->GetFunction());
-  Nan::Set(exports, Nan::New("getSmallFriendAvatar").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetSmallFriendAvatar)->GetFunction());
-  Nan::Set(
-      exports, Nan::New("getMediumFriendAvatar").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(GetMediumFriendAvatar)->GetFunction());
-  Nan::Set(
-      exports, Nan::New("getLargeFriendAvatar").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(GetLargeFriendAvatar)->GetFunction());
-  Nan::Set(
-      exports, Nan::New("requestUserInformation").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(RequestUserInformation)->GetFunction());
-  Nan::Set(exports, Nan::New("setListenForFriendsMessage").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(SetListenForFriendsMessages)
-               ->GetFunction());
-  Nan::Set(exports, Nan::New("replyToFriendMessage").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(ReplyToFriendMessage)
-               ->GetFunction());
-  Nan::Set(exports, Nan::New("getFriendMessage").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetFriendMessage)
-               ->GetFunction());
+  SET_FUNCTION("getFriendCount", GetFriendCount);
+  SET_FUNCTION("getFriends", GetFriends);
+  SET_FUNCTION("getSmallFriendAvatar", GetSmallFriendAvatar);
+  SET_FUNCTION("getMediumFriendAvatar", GetMediumFriendAvatar);
+  SET_FUNCTION("getLargeFriendAvatar", GetLargeFriendAvatar);
+  SET_FUNCTION("requestUserInformation", RequestUserInformation);
+  SET_FUNCTION("setListenForFriendsMessage", SetListenForFriendsMessages);
+  SET_FUNCTION("replyToFriendMessage", ReplyToFriendMessage);
+  SET_FUNCTION("getFriendMessage", GetFriendMessage);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_registry.h
+++ b/src/api/steam_api_registry.h
@@ -20,6 +20,11 @@
     obj->Set(Nan::New(type_name).ToLocalChecked(), \
              Nan::New(type))
 
+#define SET_FUNCTION(function_name, function)                 \
+  Nan::Set(target, Nan::New(function_name).ToLocalChecked(),          \
+           Nan::GetFunction(Nan::New<v8::FunctionTemplate>(function)) \
+               .ToLocalChecked());
+
 namespace greenworks {
 namespace api {
 

--- a/src/api/steam_api_settings.cc
+++ b/src/api/steam_api_settings.cc
@@ -265,55 +265,25 @@ NAN_METHOD(GetImageRGBA) {
           .ToLocalChecked());
 }
 
-void RegisterAPIs(v8::Local<v8::Object> exports) {
-  Nan::Set(exports,
+void RegisterAPIs(v8::Local<v8::Object> target) {
+  Nan::Set(target,
            Nan::New("_version").ToLocalChecked(),
            Nan::New(GREENWORKS_VERSION).ToLocalChecked());
-  Nan::Set(
-      exports, Nan::New("restartAppIfNecessary").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(RestartAppIfNecessary)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("isSteamRunning").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(IsSteamRunning)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getSteamId").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetSteamId)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getAppId").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetAppId)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getCurrentGameLanguage").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               GetCurrentGameLanguage)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getCurrentUILanguage").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetCurrentUILanguage)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getCurrentGameInstallDir").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               GetCurrentGameInstallDir)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getNumberOfPlayers").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetNumberOfPlayers)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("isGameOverlayEnabled").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(IsGameOverlayEnabled)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("activateGameOverlay").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(ActivateGameOverlay)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("activateGameOverlayToWebPage").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               ActivateGameOverlayToWebPage)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("isSubscribedApp").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(IsSubscribedApp)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getImageSize").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetImageSize)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("getImageRGBA").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetImageRGBA)->GetFunction());
+
+  SET_FUNCTION("restartAppIfNecessary", RestartAppIfNecessary);
+  SET_FUNCTION("isSteamRunning", IsSteamRunning);
+  SET_FUNCTION("getSteamId", GetSteamId);
+  SET_FUNCTION("getAppId", GetAppId);
+  SET_FUNCTION("getCurrentGameLanguage", GetCurrentGameLanguage);
+  SET_FUNCTION("getCurrentUILanguage", GetCurrentUILanguage);
+  SET_FUNCTION("getCurrentGameInstallDir", GetCurrentGameInstallDir);
+  SET_FUNCTION("getNumberOfPlayers", GetNumberOfPlayers);
+  SET_FUNCTION("isGameOverlayEnabled", IsGameOverlayEnabled);
+  SET_FUNCTION("activateGameOverlay", ActivateGameOverlay);
+  SET_FUNCTION("activateGameOverlayToWebPage", ActivateGameOverlayToWebPage);
+  SET_FUNCTION("isSubscribedApp", IsSubscribedApp);
+  SET_FUNCTION("getImageSize", GetImageSize);
+  SET_FUNCTION("getImageRGBA", GetImageRGBA);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_stats.cc
+++ b/src/api/steam_api_stats.cc
@@ -134,16 +134,11 @@ NAN_METHOD(ResetAllStats) {
 }
 
 void RegisterAPIs(v8::Local<v8::Object> target) {
-  Nan::Set(target, Nan::New("getStatInt").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetStatInt)->GetFunction());
-  Nan::Set(target, Nan::New("getStatFloat").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(GetStatFloat)->GetFunction());
-  Nan::Set(target, Nan::New("setStat").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(SetStat)->GetFunction());
-  Nan::Set(target, Nan::New("storeStats").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(StoreStats)->GetFunction());
-  Nan::Set(target, Nan::New("resetAllStats").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(ResetAllStats)->GetFunction());
+  SET_FUNCTION("getStatInt", GetStatInt);
+  SET_FUNCTION("getStatFloat", GetStatFloat);
+  SET_FUNCTION("setStat", SetStat);
+  SET_FUNCTION("storeStats", StoreStats);
+  SET_FUNCTION("resetAllStats", ResetAllStats);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/api/steam_api_workshop.cc
+++ b/src/api/steam_api_workshop.cc
@@ -374,40 +374,21 @@ NAN_METHOD(UGCUnsubscribe) {
   info.GetReturnValue().Set(Nan::Undefined());
 }
 
-void RegisterAPIs(v8::Local<v8::Object> exports) {
-  InitUgcMatchingTypes(exports);
-  InitUgcQueryTypes(exports);
-  InitUserUgcListSortOrder(exports);
-  InitUserUgcList(exports);
+void RegisterAPIs(v8::Local<v8::Object> target) {
+  InitUgcMatchingTypes(target);
+  InitUgcQueryTypes(target);
+  InitUserUgcListSortOrder(target);
+  InitUserUgcList(target);
 
-  Nan::Set(exports,
-           Nan::New("fileShare").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(FileShare)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("_publishWorkshopFile").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(PublishWorkshopFile)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("_updatePublishedWorkshopFile").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(
-               UpdatePublishedWorkshopFile)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("_ugcGetItems").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(UGCGetItems)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("_ugcGetUserItems").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(UGCGetUserItems)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("ugcDownloadItem").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(UGCDownloadItem)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("_ugcSynchronizeItems").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(UGCSynchronizeItems)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("ugcShowOverlay").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(UGCShowOverlay)->GetFunction());
-  Nan::Set(exports,
-           Nan::New("ugcUnsubscribe").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(UGCUnsubscribe)->GetFunction());
+  SET_FUNCTION("fileShare", FileShare);
+  SET_FUNCTION("_publishWorkshopFile", PublishWorkshopFile);
+  SET_FUNCTION("_updatePublishedWorkshopFile", UpdatePublishedWorkshopFile);
+  SET_FUNCTION("_ugcGetItems", UGCGetItems);
+  SET_FUNCTION("_ugcGetUserItems", UGCGetUserItems);
+  SET_FUNCTION("ugcDownloadItem", UGCDownloadItem);
+  SET_FUNCTION("_ugcSynchronizeItems", UGCSynchronizeItems);
+  SET_FUNCTION("ugcShowOverlay", UGCShowOverlay);
+  SET_FUNCTION("ugcUnsubscribe", UGCUnsubscribe);
 }
 
 SteamAPIRegistry::Add X(RegisterAPIs);

--- a/src/greenworks_api.cc
+++ b/src/greenworks_api.cc
@@ -47,9 +47,7 @@ NAN_MODULE_INIT(init) {
 
   greenworks::api::SteamAPIRegistry::GetInstance()->RegisterAllAPIs(target);
 
-  Nan::Set(target,
-           Nan::New("initAPI").ToLocalChecked(),
-           Nan::New<v8::FunctionTemplate>(InitAPI)->GetFunction());
+  SET_FUNCTION("initAPI", InitAPI);
 }
 
 }  // namespace

--- a/src/steam_id.cc
+++ b/src/steam_id.cc
@@ -39,7 +39,7 @@ v8::Local<v8::Object> SteamID::Create(CSteamID steam_id) {
 
   auto* obj = new SteamID(steam_id);
   v8::Local<v8::Object> instance =
-      Nan::NewInstance(tpl->GetFunction()).ToLocalChecked();
+      Nan::NewInstance(Nan::GetFunction(tpl).ToLocalChecked()).ToLocalChecked();
   Nan::SetInternalFieldPointer(instance, 0, obj);
   return scope.Escape(instance);
 }

